src/pages/publication/references_tab/BookReferenceForm.tsx

```diff
@@ -1,29 +1,207 @@
+  font-size: 1.5rem;
+  font-weight: bold;
+  grid-area: header;
+`;
 
-const BookReferenceForm: React.FC = () => {
+const StyledNviInformation = styled.div`
+  grid-area: information;
+`;
+
+const StyledCheckCircleIcon = styled(CheckCircleIcon)`
+  grid-area: icon;
+  color: green;
+  margin: 1rem;
+  font-size: 2rem;
+`;
+
+const StyledCancelIcon = styled(CancelIcon)`
+  grid-area: icon;
+  color: red;
+  margin: 1rem;
+  font-size: 2rem;
+`;
+
+const BookReferenceForm: FC = () => {
   const { t } = useTranslation('publication');
 
-  const { setFieldValue }: FormikProps<any> = useFormikContext();
+  const { setFieldValue, values }: FormikProps<PublicationFormsData> = useFormikContext();
+
+  const isRatedBook = values.reference?.book?.publisher?.level && values.reference.book.publisher.level !== '0';
+
+  const isPeerReviewed = values.reference?.book?.peerReview;
 
   return (
     <>
-      <Field name={BookFieldNames.TYPE} component={Select} variant="outlined" fullWidth>
-        {bookTypes.map(type => (
-          <MenuItem value={type.value} key={type.value}>
-            {t(type.label)}
-          </MenuItem>
-        ))}
+      <Field name={BookFieldNames.TYPE}>
+        {({ field: { onChange, name, value } }: any) => (
+          <FormControl variant="outlined" fullWidth>
+            <InputLabel>{t('common:type')}</InputLabel>
+            <Select value={value} onChange={onChange(name, value)}>
+              {bookTypes.map(type => (
+                <MenuItem value={type.value} key={type.value}>
+                  {t(type.label)}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        )}
       </Field>
 
       <Field name={BookFieldNames.PUBLISHER}>
-        {() => <PublisherSearch setValueFunction={value => setFieldValue(BookFieldNames.PUBLISHER, value)} />}
+        {({ field: { name, value } }: any) => (
+          <>
+            <PublicationChannelSearch
+              clearSearchField={value === emptyPublisher}
+              label={t('references.publisher')}
+              publicationTable={PublicationTableNumber.PUBLISHERS}
+              setValueFunction={value => setFieldValue(name, value ?? emptyPublisher)}
+            />
+            {value.title && (
+              <JournalPublisherRow
+                hidePublisher
+                label={t('references.publisher')}
+                publisher={value}
+                onClickDelete={() => setFieldValue(name, emptyPublisher)}
+              />
+            )}
+          </>
+        )}
+      </Field>
+      <Field name={BookFieldNames.ISBN}>
+        {({ field }: any) => <TextField variant="outlined" label={t('references.isbn')} {...field} />}
+      </Field>
+      <StyledSection>
+        <StyledPeerReview>
+          <Field name={BookFieldNames.PEER_REVIEW}>
+            {({ field }: any) => (
+              <PeerReview field={field} label={t('references.peer_review')} setFieldValue={setFieldValue} />
```

Comment:
hva med å sende `fieldName` i stedet for `field` og `setFieldValue` her, så kan vi ha `<PeerReviewField />`som oppretter Field med riktig name og setter verdi via useFormikContext()?

Reply:
ja, tenkte på det jeg også. kan gjøre det 👍